testkit/src/main/java/org/apache/calcite/test/DiffRepository.java

```diff
@@ -255,6 +255,15 @@ private DiffRepository(URL refFile, File logFile,
   //~ Methods ----------------------------------------------------------------
 
   public void checkActualAndReferenceFiles() {
+    // Check if any test cases are out of order FIRST, before other checks
+    // This ensures that we fail fast if there are ordering issues
+    if (!outOfOrderTests.isEmpty()) {
+      throw new IllegalArgumentException("TestCase(s) are out of alphabetical order in the"
+          + " reference file: " + Sources.of(refFile).file() + "\n"
+          + "Out-of-order test cases: " + outOfOrderTests + "\n"
+          + "To fix, copy the generated log file: " + logFile);
+    }
+
```

Comment:
Why do we need this check here? Aren't the remaining checks sufficient to detect that the files are not identical? Can't we follow the same pattern with the check below:
```java
if(existsMethodOnlyInXml || !outOfOrderTests.isEmpty()) {
  modCount++;
  flushDoc();
}
```